app/views/donations/new.html.erb

```diff
@@ -0,0 +1,88 @@
+<header>
+  <h1><%= t 'views.donations.support.heading' %></h1>
+</header>
+
+<% if flash[:error].present? %>
+  <div id='error_explanation'>
+    <p><%= flash[:error] %></p>
+  </div>
+<% end %>
+
+<div class="row">
+  <div class='column column-one-third'>
+    <%= form_tag [:support], id: 'js-stripe-form' do %>
+      <script src='https://checkout.stripe.com/checkout.js'></script>
+
+      <!-- These get populated by the token callback below -->
+      <%= hidden_field_tag 'js-stripe-email' %>
+      <%= hidden_field_tag 'js-stripe-token' %>
+
+      <div class='row buttons'>
+        <div class='column column-half'>
+          <%= radio_button_tag :monthly, true, checked: true %>
+          <%= label_tag :monthly_true, t('views.donations.support.radio_label_monthly') %>
+        </div>
+
+        <div class='column column-half'>
+          <%= radio_button_tag :monthly, false %>
+          <%= label_tag :monthly_false, t('views.donations.support.radio_label_one_time') %>
+        </div>
+      </div>
+
+      <%= range_field_tag 'amount_slider', 50, in: 1..1000, step: 1 %>
+
+      <%= label_tag :amount, t('views.donations.support.amount_label') %>
+      <%= text_field_tag :amount, '$50', tabindex: '-1' %>
```

Comment:
this feedback may be overkill for the initial implementation, so feel free to defer for now:

this is making me nervous. Does strip have a frontend UI library?

If not, maybe make it a `<input type="number" step="1" min="0" max="10000">` and use javascript (Jquery or vanilla) to make sure the backend just gets an integer value in cents and a currency code

Reply:
I have a solution to this that I am just wrapping up right now.

It cleans up some of the js where we were adding and removing $ and sets the field to a number field. It did require that I move the "amount" to being left aligned. It was much easier that way and I think it's a cleaner implementation. Pushing it up in a sec.